### PR TITLE
week3-js-homework Abdulrahman sadek

### PR DESCRIPTION
**Pull request by Abdulrahmansadek**

### CODE REVIEW

File: 1-JavaScript/Week3/homework/ex1-giveCompliment.js [hunk 1]
```diff
@@ -18,11 +18,24 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/1-Java
...
+    'awesome',
+    'excellent',
+    'impressive',
+    'gorgeous',
+    'wonderful',
+    'brilliant',
+    'sharp',
+  ];
+  return `You are ${
+    compliments[Math.floor(Math.random() * compliments.length)]
```
Comment (yash-kapila) [inline 1-JavaScript/Week3/homework/ex1-giveCompliment.js:35]: 👍

File: 1-JavaScript/Week3/homework/ex2-dogYears.js [hunk 2]
```diff
@@ -16,8 +16,9 @@ calculate it!
    ages.
 -----------------------------------------------------------------------------*/
 
-function calculateDogAge(/* parameters go here */) {
-  // TODO complete this function
+function calculateDogAge(humanYear) {
+  const dogYear = humanYear * 7;
+  return `Your doggie is ${dogYear} years old in dog years!`;
```
Comment (yash-kapila) [inline 1-JavaScript/Week3/homework/ex2-dogYears.js:21]: 👍

File: 1-JavaScript/Week3/homework/ex3-tellFortune.js [hunk 3]
```diff
@@ -30,30 +30,28 @@ Note: The DRY principle is put into practice here: instead of repeating the code
...
+function selectRandomly(arr) {
+  const random = arr[Math.floor(Math.random() * arr.length)];
+
+  return random;
 }
 
-function tellFortune(/* add parameter(s) here */) {
-  // TODO complete this function
+function tellFortune(numKids, partnerNames, locations, jobTitles) {
+  return `You will be a ${selectRandomly(jobTitles)} in ${selectRandomly(
```
Comment (yash-kapila) [inline 1-JavaScript/Week3/homework/ex3-tellFortune.js:50]: 👍

File: 1-JavaScript/Week3/homework/ex4-shoppingCart.js [hunk 4]
```diff
@@ -23,8 +23,13 @@ you have more than 3 items in your shopping cart the first item gets taken out.
...
 
 // ! Function to be tested
-function addToShoppingCart(/* parameters go here */) {
-  // TODO complete this function
+function addToShoppingCart(groceryItem) {
+  shoppingCart.push(groceryItem);
+  if (shoppingCart.length > 3) {
+    shoppingCart.shift();
+  }
+
```
Comment (yash-kapila) [inline 1-JavaScript/Week3/homework/ex4-shoppingCart.js:31]: Might also be worth adding a small logic here to check if `groceryItem` is defined or not

File: 1-JavaScript/Week3/homework/ex5-shoppingCartPure.js [hunk 5]
```diff
@@ -16,13 +16,19 @@ it pure. Do the following:
 5. Confirm that you function passes the provided unit tests.
 ------------------------------------------------------------------------------*/
 // ! Function under test
-function addToShoppingCart(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function addToShoppingCart(groceryItem, shoppingCart) {
+  shoppingCart = ['bananas', 'milk'];
+  const newShoppingCart = [...shoppingCart, groceryItem];
+
+  if (newShoppingCart.length > 3) {
```
Comment (yash-kapila) [inline 1-JavaScript/Week3/homework/ex5-shoppingCartPure.js:23]: Same change requested as above

File: 1-JavaScript/Week3/homework/ex6-totalCost.js [hunk 6]
```diff
@@ -21,20 +21,28 @@ instead!
...
+  water: 1.7,
 };
 
-function calculateTotalPrice(/* TODO parameter(s) go here */) {
-  // TODO replace this comment with your code
+function calculateTotalPrice(cart) {
+  let totalValue = 0;
+  for (const grocery of Object.values(cart)) {
+    totalValue += grocery;
+  }
```
Comment (yash-kapila) [inline 1-JavaScript/Week3/homework/ex6-totalCost.js:35]: Would suggest to explore refactoring this logic using `Array.prototype.reduce` instead

File: 1-JavaScript/Week3/homework/ex7-mindPrivacy.js [hunk 7]
```diff
@@ -30,18 +30,22 @@ const employeeRecords = [
...
 
 // ! Function under test
-function filterPrivateData(/* parameter(s) go here */) {
-  // TODO complete this function
+function filterPrivateData(employeeRecords) {
+  const privacy = employeeRecords.map((employee) => {
+    const { name, occupation, email } = employee;
+    return { name, occupation, email };
+  });
+  return privacy;
```
Comment (yash-kapila) [inline 1-JavaScript/Week3/homework/ex7-mindPrivacy.js:38]: 👍

**Review by yash-kapila — APPROVED** [review]
Hi @Abdulrahmansadek - I have finished reviewing the homework. Please have a look at the suggested changes.

File: 1-JavaScript/Week3/homework/ex1-giveCompliment.js (hunk 1 above)
Comment (Abdulrahmansadek) [reply 1-JavaScript/Week3/homework/ex1-giveCompliment.js:35]: Thanks @yash-kapila for reviewing my homework .

**Pull request closed without merging**